code/game/objects/items/weapons/lighters.dm

```diff
@@ -118,19 +124,22 @@
 	return
 
 /obj/item/lighter/update_icon_state()
-	icon_state = "[initial(icon_state)][lit ? "-on" : ""]"
+	icon_state = "[base_icon_state ? "[base_icon_state]" : "[initial(icon_state)]"][lighter_color ? "-[lighter_color]" : ""][lit ? "-on" : ""]"
```

Comment:
I think this is what you're looking for, and do the same thing for the item_state
```suggestion
	icon_state = "[base_icon_state][lighter_color ? "-[lighter_color]" : ""][lit ? "-on" : ""]"
```